types/src/proof/definition.rs

```diff
@@ -426,6 +428,48 @@ impl From<AccumulatorConsistencyProof> for crate::proto::types::AccumulatorConsi
     }
 }
 
+/// A proof that first verifies that establishes correct computation of the root and then
+/// returns the new tree to acquire a new root and version
+#[derive(Deserialize, Serialize)]
+pub struct AccumulatorExtensionProof<H> {
```

Comment:
I think you are re-implementing `AccumulatorConsistencyProof` created by @wqfish . You guys should talk. Basically the takeaway from our discussion about this proof is it can just be a wrapper of `Vec<HashValue>`

Reply:
Not quite. We require a superset of what the AccumulatorConsistencyProof provides. The ConsistencyProof assumes you have a tree that you want to extend. The TCB only has the original root hash. We need proof to compute it correctly and then proof that it is extended correctly. We could theoretically make this proof contain an AccumulatorConsistencyProof but I think that adds a lot of mental complexity. FWIW, we did chat and this was the (high-level) agreed upon approach. Arguably the implementation may or may not be what he had in mind.

Reply:
Discussion was to move this to consensus-types since it is used internally only.